build.gradle

```diff
@@ -524,7 +509,25 @@ task sparkJar(type: ShadowJar) {
     }
 }
 
-task bundle(type: Zip) {
+// A jar that only contains the test classes and resources (to be extracted for testing)
+task shadowTestClassJar(type: ShadowJar){
+    group = "Shadow"
```

Comment:
Add description lines to the new tasks so that people who see them have some idea what they're for.